server/src/main/java/org/opensearch/identity/PrincipalIdentifierToken.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity;
+
+import org.opensearch.common.io.stream.NamedWriteable;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Requester Token for requests to/from an extension
+ * Each user will have different token for different extension
+ */
+public class PrincipalIdentifierToken implements Serializable, NamedWriteable {
+    public static final String NAME = "principal_identifier_token";
+    private String token;
+
+    public PrincipalIdentifierToken(String token) {
```

Comment:
Please add a javadoc here explaining what type of String is expected.  Is this a JWT?  Any String?  I see elsewhere in `extractToken()` it is expected to have two fields around a colon, but don't JWTs have two colons?  

Consider throwing an `IllegalArgumentException` for malformed tokens here, so you don't get the exception on `getToken()` in other code.

Reply:
For now the input is a simple colon-separated String, but once Encryption/Decryption is implemented we can update the documentation here. +1 to throwing exceptions part.